source/agora/node/Registry.d

```diff
@@ -611,45 +596,63 @@ private struct ZoneData
+    public this (string zone_name, Domain root, ZoneConfig config,
+        ManagedDatabase cache_db, Logger logger)
     {
         this.db = cache_db;
         this.log = logger;
+        this.config = config;
+        this.root = root;
+
+        static string serverType (bool auth)
+        {
+            return auth ? "authoritative" : "secondary";
+        }
+
+        this.log.info("Registry is {} DNS server for zone '{}'",
+            serverType(this.config.authoritative), this.root);
 
         this.query_count = format("SELECT COUNT(*) FROM registry_%s_signature",
-            type_table_name);
+            zone_name);
 
         this.query_registry_get = format("SELECT pubkey " ~
-            "FROM registry_%s_signature", type_table_name);
+            "FROM registry_%s_signature", zone_name);
 
         this.query_payload = format("SELECT signature, sequence, address, type, utxo " ~
             "FROM registry_%s_addresses l " ~
             "INNER JOIN registry_%s_signature r ON l.pubkey = r.pubkey " ~
-            "WHERE l.pubkey = ?", type_table_name, type_table_name);
+            "WHERE l.pubkey = ?", zone_name, zone_name);
 
         this.query_signature_add = format("REPLACE INTO registry_%s_signature " ~
-            "(pubkey, signature, sequence, utxo) VALUES (?, ?, ?, ?)", type_table_name);
+            "(pubkey, signature, sequence, utxo) VALUES (?, ?, ?, ?)", zone_name);
 
         this.query_addresses_add = format("REPLACE INTO registry_%s_addresses " ~
-                    "(pubkey, address, type) VALUES (?, ?, ?)", type_table_name);
+                    "(pubkey, address, type) VALUES (?, ?, ?)", zone_name);
 
         this.query_remove_sig = format("DELETE FROM registry_%s_signature WHERE pubkey = ?",
-            type_table_name);
+            zone_name);
 
         this.query_addresses_get = format("SELECT address " ~
-            "FROM registry_%s_addresses", type_table_name);
+            "FROM registry_%s_addresses", zone_name);
 
         string query_sig_create = format("CREATE TABLE IF NOT EXISTS registry_%s_signature " ~
             "(pubkey TEXT, signature TEXT NOT NULL, sequence INTEGER NOT NULL, " ~
-            "utxo TEXT NOT NULL, PRIMARY KEY(pubkey))", type_table_name);
+            "utxo TEXT NOT NULL, PRIMARY KEY(pubkey))", zone_name);
 
         string query_addr_create = format("CREATE TABLE IF NOT EXISTS registry_%s_addresses " ~
             "(pubkey TEXT, address TEXT NOT NULL, type INTEGER NOT NULL, " ~
             "FOREIGN KEY(pubkey) REFERENCES registry_%s_signature(pubkey) ON DELETE CASCADE, " ~
-            "PRIMARY KEY(pubkey, address))", type_table_name, type_table_name);
+            "PRIMARY KEY(pubkey, address))", zone_name, zone_name);
 
         this.db.execute(query_sig_create);
         this.db.execute(query_addr_create);
+
+        // Serial's value wraps around so the cast is safe
+        const serial = cast(uint) Clock.currTime(UTC()).toUnixTime();
```

Comment:
Note that the reason why `serial` was passed down was so that they would both have the same serial. However, considering it again, I'm not sure it really is a problem.

Reply:
`SERIAL` is a way to detect data changes, since these are separate zones I don't think this would be a problem